### PR TITLE
Add Effector typings

### PR DESCRIPTION
**Pull request by igorkamyshev**
Reopened #4372 

- Links to documentation: https://effector.dev
- Link to GitHub or NPM: https://github.com/effector/effector
- Type of contribution: new definition

Other notes:

Hello, folks. I'm Igor, Effector Core Team member. We are preparing the new major release (v23) and have decided to move Flow-typings from the main repo of Effector to this repository.

Please consider merging my PR.

Tracking issue in Effector — https://github.com/effector/effector/issues/755

### CODE REVIEW

**Comment by igorkamyshev** [comment]
Hello, @Brianzchen!

Thanks for your review in previous PR. I have fixed directory structure, rename branch and add some cases for basic scenario.

Could you check it one more time, please?

File: definitions/npm/effector-react_v22.x.x/flow_v0.83.x-/effector-react_v22.x.x.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
+declare module 'effector-react' {
+  //@flow strict
```
Comment (Brianzchen) [inline definitions/npm/effector-react_v22.x.x/flow_v0.83.x-/effector-react_v22.x.x.js:2]: don't need this at all in the file. But also flow pragmas should only be referenced at the top of the file. I imagine if you removed this everything should still work

File: definitions/npm/effector-react_v22.x.x/flow_v0.83.x-/effector-react_v22.x.x.js [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+  import type {
+    ComponentType,
+    Component,
+    ElementConfig,
+    Node,
+    Context as ContextType,
+  } from 'react'
+  import type {Store, Event} from 'effector'
+
+  export type StoreConsumer<State> = ComponentType<{|
```
Comment (Brianzchen) [inline definitions/npm/effector-react_v22.x.x/flow_v0.83.x-/effector-react_v22.x.x.js:13]: All these exports are missing the `declare` keyword which is necessary inside module declarations.

File: definitions/npm/effector_v22.x.x/flow_v0.83.x-/effector_v22.x.x.js [hunk 3]
```diff
@@ -0,0 +1,636 @@
+declare module 'effector' {
+  //@flow strict
```
Comment (Brianzchen) [inline definitions/npm/effector_v22.x.x/flow_v0.83.x-/effector_v22.x.x.js:2]: can remove this too

**Review by Brianzchen — CHANGES_REQUESTED** [review]

**Comment by Brianzchen** [comment]
Thank you @igorkamyshev. Since these are definitions for your own library so as long as basic standards are maintained I'm happy to merge.

You've just got a few minor syntax failures. To solve this, you can run `./quick_run_def_tests.sh` in the root to test locally. But before this you want to build the the cli under `main` with `yarn && yarn build` in `/cli`

**Comment by sergeysova** [comment]
@igorkamyshev Why you closed this?

**Comment by igorkamyshev** [comment]
> @igorkamyshev Why you closed this?

Discussed it in private conversation with effector team.

**Pull request closed without merging**